pkg/operator/controllers/subnets/subnet_serviceendpoint.go

```diff
@@ -12,49 +12,61 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 )
 
 func (r *reconcileManager) ensureSubnetServiceEndpoints(ctx context.Context, s subnet.Subnet) error {
-	subnetObject, err := r.subnets.Get(ctx, s.ResourceID)
-	if err != nil {
-		return err
-	}
+	if !gatewayEnabled(r.instance) {
```

Comment:
can we invert this logic to return early instead of nesting the existing logic?

Reply:
I originally wrote it in that "return early" way, but I changed it to the way it is now after Caden pointed out that `!gatewayEnabled` is the exception and not the norm, and so it makes sense to have that logic nested. Having considered both ways, I think I prefer the way it is now.